apps/researcher/src/app/[locale]/objects/[id]/(provenance)/sort-events.ts

```diff
@@ -1,73 +1,19 @@
 import {ProvenanceEvent} from '@colonial-collections/api';
+import {ProvenanceEventEnrichment} from '@colonial-collections/enricher';
 
-export function sortEvents(events: ProvenanceEvent[]): ProvenanceEvent[] {
-  const graph = new Map();
-  const incomingEdgeCount = new Map();
+export function sortEvents(
+  events: (ProvenanceEvent | ProvenanceEventEnrichment)[]
+) {
+  return events.sort((a, b) => {
+    // If startDate is missing, consider it as the smallest possible value
+    const aStart = a.date?.startDate?.getTime() || -Infinity;
+    const bStart = b.date?.startDate?.getTime() || -Infinity;
 
-  // Step 1: Initialize the `graph` and `incomingEdgeCount` map
-  // Input: events array.
-  // Operation: For each event, create an empty array in the graph,
-  // and set the incoming edge count to 0.
-  // Output: `graph` and `incomingEdgeCount` maps initialized with each event id.
-  events.forEach(event => {
-    graph.set(event.id, []);
-    incomingEdgeCount.set(event.id, 0);
-  });
+    // If endDate is missing, consider it as the smallest possible value
```

Comment:
Or is it the highest possible value? If an end date is missing then, semantically, the event is still ongoing.

Reply:
This is true; I will change it.